File: Computer-Seekho-.NET/Computer_Seekho_DN/Models/Placement.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Computer_Seekho_DN.Models;

[Table("placement")]
[Index("BatchId", Name = "FKfjp90ouyl79cs2u450vca7ip9")]
[Index("RecruiterId", Name = "FKmeq262aa8o8iqkv0m6jpj9wog")]
[Index("StudentId", Name = "UKburs792scucffx16rarw5dijl", IsUnique = true)]
public partial class Placement
{
    [Key]
    [Column("placement_id")]
    public int PlacementId { get; set; }

    [Column("batch_id")]
    public int? BatchId { get; set; }

    [Column("recruiter_id")]
    public int? RecruiterId { get; set; }

    [Column("student_id")]
    public int? StudentId { get; set; }

    [ForeignKey("BatchId")]
    [InverseProperty("Placements")]
    public virtual Batch? Batch { get; set; }

    [ForeignKey("RecruiterId")]
    [InverseProperty("Placements")]
    public virtual Recruiter? Recruiter { get; set; }

    [ForeignKey("StudentId")]
    [InverseProperty("Placement")]
    public virtual Student? Student { get; set; }
}

File: Computer-Seekho-Java/ComputerSeekho/src/main/java/com/Project/Entities/Placement.java
package com.Project.Entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Data
@Table(name = "placement")
public class Placement {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "placement_id")
    private int placementID;

    @OneToOne
    @JoinColumn(name = "student_id", referencedColumnName = "student_id")
    private Student studentID;

    @ManyToOne
    @JoinColumn(name = "recruiter_id", referencedColumnName = "recruiter_id")
    private Recruiter recruiterID;

    @ManyToOne
    @JoinColumn(name = "batch_id",referencedColumnName = "batch_id")
    private Batch batch;
}
